doc/CI_Hardware.md

```diff
@@ -0,0 +1,243 @@
+  * [Configuration Files](#configuration-files)
+  * [Looking in the Workflow ([tock-hw-ci.yml](https://github.com/goodoomoodoo/tock/blob/master/.github/workflows/tock-hw-ci.yml))](#looking-in-the-workflow-tock-hw-ciymlhttpsgithubcomgoodoomoodootockblobmastergithubworkflowstock-hw-ciyml)
+- [Where Tests are Located and How They Work](#where-tests-are-located-and-how-they-work)
+  * [Location](#location)
+  * [How Tests Work (Example - [gpio folder](https://github.com/goodoomoodoo/libtock-c/tree/master/examples/ci-tests/gpio))](#how-tests-work-example---gpio-folderhttpsgithubcomgoodoomoodoolibtock-ctreemasterexamplesci-testsgpio)
+- [How Tests are Chosen](#how-tests-are-chosen)
+- [What to do if Something Goes Wrong](#what-to-do-if-something-goes-wrong)
+  * [If a test fails](#if-a-test-fails)
+  * [If there’s boot up issues of the Raspberry Pi](#if-theres-boot-up-issues-of-the-raspberry-pi)
+
+<!-- tocstop -->
+
+![Workflow](tock/doc/images/workflow.png)
+
+# How To Set Up a New Instance
+
+## Getting Started ([Source](https://goodoomoodoo.github.io/tock-test-harness/GUIDE))
+
+1. Install Ubuntu Server on Raspberry Pi.
+    * Follow the guide [here](https://ubuntu.com/tutorials/how-to-install-ubuntu-on-your-raspberry-pi#1-overview) until step 5. We do not need desktop environment.
+2. Install the [Requirements](https://github.com/tock/tock/blob/master/doc/Getting_Started.md#requirements) here.
+3. Download and install JLinkExe
+```bash
+$ wget --post-data 'accept_license_agreement=accepted&non_emb_ctr=confirmed&submit=Download+software' https://www.segger.com/downloads/jlink/JLink_Linux_arm64.tgz
+$ tar xvf JLink_Linux_arm64.tgz
+$ sudo cp JLink_Linux_V700a_arm64/99-jlink.rules /etc/udev/rules.d/ # Depends on JLink version
+ 
+# Add the Jlink directory to the path in .profile or .bashrc
+```
+4. Install pip3, if not yet installed
+```bash
+sudo apt install python-pip3
+```
+5. Checkout test harness in home directory 
+```bash
+cd ~; git clone https://github.com/goodoomoodoo/tock-test-harness.git
+cd ~; git clone https://github.com/goodoomoodoo/libtock-c.git
+```
+6. Install gpiozero if not yet installed. (It should come with the image)
+```bash
+$ sudo apt install python3-gpiozero
+$ sudo chown root:$USER /dev/gpiomem
+$ sudo chmod g+rw /dev/gpiomem
+```
+7. Install linux library
+```bash
+$ sudo apt install arm-none-eabi
+# Install bluetooth library
+$ sudo apt-get install bluez bluez-tools pi-bluetooth 
+# !IMPORTANT! Remember to reboot after installation
+```
+8. Install python3 libraries
+```bash
+$ cd ~/tock-test-harness; pip3 install -r requirements.txt
+```
+9. Hook up Action Runner of the Tock Repo to the Raspberry Pi.
+    * Go to \'Settings\' and go to \'Actions\' down the list on the left
+    * Under \'Self-hosted runners\' click **Add runner**
+    * Follow the steps to set up action runner on Raspberry Pi
+
+10. Add path to JLink to ```.path``` in Action Runner
+```bash
+$ cd ~/actions-runner; vi .path
+# Add the path
```

Comment:
Since we are controlling the whole setup here, can we just give the exact command. In fact, can we skip the editor step, replace this with something like:

`echo "JLINK_PATH_WHATEVER_IT_IS" > ~/actions-runner/.path`

?